tingbot/graphics.py

```diff
@@ -217,6 +217,51 @@ def text(self, string, xy=None, color='grey', align='center', font=None, font_si
 
         self.image(text_image, xy=xy, align=align, scale=1)
 
+    def oval(self, xy=None, size=(100,100), color='grey', align='center'):
+        """
+        Draws an oval.
+        
+        Args:
+            xy (tuple): The position (x, y) to draw the oval, as measured from the top-left.
+            size (tuple): The size (width, height) of the oval.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the oval relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        if len(size) != 2:
+            raise ValueError('size should be a 2-tuple')
+
+        xy = _topleft_from_aligned_xy(xy, align, size, self.size)
+
+        pygame.draw.ellipse(self.surface, _color(color), pygame.Rect(xy, size), 0) 
+
+    def circle(self, xy=None, size=100, color='grey', align='center'):
+        """
+        Draws a circle.
+
+        Args:
+            xy (tuple): The position (x, y) to draw the circle, as measured from the top-left.
+            size (int): The diameter of the circle.
+            color (tuple or str): The color (r, g, b) or color name.
+            align (str): How to align the outside box of the circle relative to `xy`, or relative to the drawing surface
+                if `xy` is None. Defaults to 'center'.
+        """
+        if xy is None:
+            raise ValueError('no value given for xy')
+        elif type(xy) is not tuple:
+            raise ValueError('xy should be a 2-tuple')
+        elif len(xy) != 2:
+            raise ValueError('xy should be a 2-tuple')
```

Comment:
I'd argue that you shouldn't be testing directly for whether xy is a tuple or not - what if a `list` was passed? or a `NamedTuple`?
My preferred solution would be to not even test this at all - if xy allows xy[0] and xy[1], then it will just work, or you will get an exception raised later on.
If you really feel the need to check the type may I suggest:
`
elif not isinstance(xy,collections.Sequence):
`
This will allow through anything based on collections.Sequence, which includes lists, tuples, NamedTuples etc.